marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/QueryStepRunner.java

```diff
@@ -129,7 +136,31 @@ public StepRunner withStopOnFailure(boolean stopOnFailure) {
 
     @Override
     public StepRunner withOptions(Map<String, Object> options) {
-        this.options = options;
+        if(flow == null){
```

Comment:
Unit tests would be easy to write for this, and this seems important to test as well - and I guess the aggregation of options from the incoming map, the flow, and the step is needed for the "disable job output" flag to work? If so, the assertion messages in unit tests are a good place to document the "why" behind this code addition.